### ISSUE TITLE
Transition to a parallel / child state throws a "Cannot read property 'stateNode' of undefined"

### ISSUE THREAD

**Issue by DirkWolthuis** [davidkpiano/xstate#520]
## Description:
If I want to transition a external state node, I expect to call it by its ID (#) and call a state with the dot notation. But if I use ` target: "#feedback.showing"` xstate throws a error. (Cannot read property 'stateNode' of undefined). What am I doing wrong? 

```
export const checkoutStateMachine = Machine(
    {
        id: "checkout",
        type: "parallel",
        context: {
            selectedDeliveryMethod: "DELIVERY",
            expressItems: []
        },
        states: {
            sections: {
                initial: "deliverySection",
                states: {
                    deliverySection: {
                        on: {
                            NEXT: [
                                {
                                    target: "invoiceSection",
                                    cond: "isDeliverySectionValid"
                                },
                                {
                                    target: "#feedback.showing",
                                }
                            ]
                        },
                  ...REDACTED
            },
            feedback: {
                id: "feedback",
                initial: "hidden",
                states: {
                    hidden: {
                        on: {
                            SHOW: "showing"
                        }
                    },
                    showing: {
                        on: {
                            HIDE: "hidden"
                        }
                    }
                }
            }
        }
    },

);```

**Comment by davidkpiano** [comment]
Related to #518, fix for this is coming 🔜

**Comment by DirkWolthuis** [comment]
Ok thanks @davidkpiano  for letting me know. I will check the releases to see if the the bug is fixed.

**Comment by davidkpiano** [comment]
@DirkWolthuis this should actually be fixed in the latest version, can you try there real quick?

Try creating a [CodeSandbox (XState template)](https://codesandbox.io/s/xstate-example-template-qcwed) reproduction of this and see if it works.

**Comment by DirkWolthuis** [comment]
@davidkpiano Here is my example (https://codesandbox.io/embed/xstate-statenode-error-520-rd0z9), tried 4.6.4 in my own project, but I get the same error. I'm probably doing it wrong :(

**Comment by davidkpiano** [comment]
Sorry, I meant the version in master. I'll verify that it's working there.

**Comment by DirkWolthuis** [comment]
Ah sorry, check.

**Comment by DirkWolthuis** [comment]
@davidkpiano Did you try it with the master branch?

**Comment by davidkpiano** [comment]
@DirkWolthuis I just released 4.6.7 with the fix. Can you update XState and try it?
